eng/common/pipelines/templates/steps/git-push-changes.yml

```diff
@@ -41,7 +41,7 @@ steps:
 
 - task: PowerShell@2
   displayName: Push changes
-  condition: and(succeeded(), eq(variables['HasChanges'], 'true'))
+  condition: and(succeeded(), ne(variables['HasChanges'], 'false'))
```

Comment:
The change is to push changes in two cases:
1. SkipCheckingForChanges is true, then we allow to push changes.
2. SkipCheckingForChanges is false, then we allow to push changes once there are any changes.
To prevent push changes:
1. SkipCheckingForChange is false, and no changes the the branch. Then skip git push step.

Reply:
After a second thought. This actually might introduce some surprise to prod. Better to have another parameter for this purpose